### PR TITLE
ProjectSetUp

### PR DESCRIPTION
**Pull request by AKSHAYAPALCHAMY-P**
ADDED: FolderFiles

### CODE REVIEW

File: AyraaCollections.Common/DataTransferObjects/Authentication/UserDTO.cs [hunk 1]
```diff
@@ -0,0 +1,18 @@
+﻿using System;
+
+namespace AyraaCollections.Common
+{
+
```
Comment (ManiRaj-021199) [inline AyraaCollections.Common/DataTransferObjects/Authentication/UserDTO.cs]: Remove unneccessary spaces.

File: AyraaCollections.Common/DataTransferObjects/Authentication/UserDTO.cs [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+    public class UserDTO
+    {
+        public Guid Id { get; set; }
+        public string Email { get; set; }
+        public string FirstName {  get; set; }
+        public string LastName { get; set; }
+        public string Password {  get; set; }
+        public string ConfirmPassword {  get; set; }     
+    }
+
```
Comment (ManiRaj-021199) [inline AyraaCollections.Common/DataTransferObjects/Authentication/UserDTO.cs]: Remove unneccessary spaces.

File: AyraaCollections.Common/Enums/Authentication/UserType.cs [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+
+namespace AyraaCollections.Common
+{
+    public enum UserType
+    {
+        Admin,
+        User
+    }
+}
+
```
Comment (ManiRaj-021199) [inline AyraaCollections.Common/Enums/Authentication/UserType.cs:11]: Remove unneccessary spaces.

File: AyraaCollections.Data/Entities/User.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+    public class User
+    {
+
+            public Guid Id { get; set; }
+            public string Email { get; set; }
+            public string FirstName { get; set; }
+            public string LastName { get; set; }
+            public string Password { get; set; }
+
+            public User() { }
```
Comment (ManiRaj-021199) [inline AyraaCollections.Data/Entities/User.cs]: Why this constructor? Is this required?

File: AyraaCollections.Data/Entities/User.cs
Comment (ManiRaj-021199) [inline AyraaCollections.Data/Entities/User.cs:1]: Use proper indendation for all lines.

File: AyraaCollections.Data/Entities/User.cs [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+            FirstName = user.FirstName;
+            LastName = user.LastName;
+            Password = user.Password;
+        }
+
+        public UserDTO GetUserDTO()
+        {
+            return new UserDTO()
+            {
+                Email = this.Email,
```
Comment (ManiRaj-021199) [inline AyraaCollections.Data/Entities/User.cs:27]: Don't you need UserID?

File: AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs [hunk 6]
```diff
@@ -0,0 +1,26 @@
+﻿using System.Linq;
+using AyraaCollections.Common;
+using AyraaCollections.Data;
+
+public class AuthenticationBL
+{
+    public bool Register(UserDTO dtoUser)
+    {
+
```
Comment (ManiRaj-021199) [inline AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs:15]: Remove unneccessary spaces.

File: AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs [hunk 7]
```diff
@@ -0,0 +1,26 @@
...
+        User user = new User(dtoUser);
```
Comment (ManiRaj-021199) [inline AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs:16]: Check user is there or not already - using email?

**Review by ManiRaj-021199 — CHANGES_REQUESTED** [review]
Please fix the comments.

Thank You.

File: AyraaCollections.Common/DataTransferObjects/Authentication/UserDTO.cs (hunk 1 above)
Comment (AKSHAYAPALCHAMY-P) [reply AyraaCollections.Common/DataTransferObjects/Authentication/UserDTO.cs]: changed

File: AyraaCollections.Common/DataTransferObjects/Authentication/UserDTO.cs (hunk 2 above)
Comment (AKSHAYAPALCHAMY-P) [reply AyraaCollections.Common/DataTransferObjects/Authentication/UserDTO.cs]: Changed

File: AyraaCollections.Data/Entities/User.cs
Comment (AKSHAYAPALCHAMY-P) [reply AyraaCollections.Data/Entities/User.cs:1]: Changed

File: AyraaCollections.Data/Entities/User.cs (hunk 4 above)
Comment (AKSHAYAPALCHAMY-P) [reply AyraaCollections.Data/Entities/User.cs]: Yes,its for table context.

File: AyraaCollections.Data/Entities/User.cs (hunk 5 above)
Comment (AKSHAYAPALCHAMY-P) [reply AyraaCollections.Data/Entities/User.cs:27]: Changed.

File: AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs (hunk 6 above)
Comment (AKSHAYAPALCHAMY-P) [reply AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs:15]: changed.

File: AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs (hunk 7 above)
Comment (AKSHAYAPALCHAMY-P) [reply AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs:16]: changed.

File: AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs [hunk 8]
```diff
@@ -0,0 +1,32 @@
+﻿using System.Linq;
+using AyraaCollections.Common;
+using AyraaCollections.Data;
+
+public class AuthenticationBL
+{
+    public bool Register(UserDTO dtoUser)
+    {
+        var ExistingUser = AyraaCollectionsContext.Users.Exists(u => u.Email == dtoUser.Email);
```
Comment (ManiRaj-021199) [inline AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs]: naming convention.? boolean. use bIsUserExist

File: AyraaCollections.Data/Entities/User.cs (hunk 4 above)
Comment (ManiRaj-021199) [reply AyraaCollections.Data/Entities/User.cs]: I don't think it required. Please check again.

**Review by ManiRaj-021199 — CHANGES_REQUESTED** [review]
Added comments.

File: AyraaCollections.Common/Enums/Authentication/UserType.cs (hunk 3 above)
Comment (AKSHAYAPALCHAMY-P) [reply AyraaCollections.Common/Enums/Authentication/UserType.cs:11]: changed.

File: AyraaCollections.Data/Entities/User.cs (hunk 4 above)
Comment (AKSHAYAPALCHAMY-P) [reply AyraaCollections.Data/Entities/User.cs]: Changed.

File: AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs (hunk 8 above)
Comment (AKSHAYAPALCHAMY-P) [reply AyraaCollections.Logic/BusinessLogics/AuthenticationBL.cs]: changed.

**Review by ManiRaj-021199 — APPROVED** [review]
Looks Good!

**Pull request merged**